src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -73,6 +73,36 @@ public static void ResetColor()
             }
         }
 
+        public static int WindowWidth
+        {
+            get
+            {
+                if (!ConsoleOutIsTerminal) return 0; // Should we rather throw exception here? On windows we get IOException from the native code, do we need to match for code portability?
```

Comment:
On windows, if the handle is redirected the native API that we use for this would INVALID_HANDLE. However, we try to use all the 3 handles, input, output and error to get the information. On Unix, we might not have to do this actually. I am yet to test the o/p of Ioctl but based on its o/p I will change this.